### PR TITLE
Add adamR script

### LINKED ISSUE #1649: Need adamR script
À la `adam-shell`, `pyadam`. Forgot to add this in #1397.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1649. I took the liberty of rewriting the `bin` scripts to make them DRYer, and added tests that Jenkins will run for `adam-shell` and `pyadam`.

### CODE REVIEW

**Comment by fnothaft** [comment]
This runs OK on my side. Anyone else able to reproduce/build?

**Comment by heuermh** [comment]
This works for me locally and locally via distribution tarball.  The problem I ran into with the conda build was that the tarball was installed in one place and bin scripts were symlinked from a different location, so the relative source directory checks weren't correct.

https://github.com/bioconda/bioconda-recipes/blob/master/recipes/adam/build.sh#L11

It's not obvious to me looking at these changes whether it will still work, and I'm not entirely sure how to test the conda build without a release tarball.  Will look into it.

**Comment by fnothaft** [comment]
Thanks for confirming @heuermh! @AmplabJenkins is having some transient issues this AM, so I'm going to wait for a resolution to that (just pinged @shaneknapp) before kicking the wheels on this PR again.

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by heuermh** [comment]
I simulated the conda build, and `adam-submit` and `adam-shell` work ok.
```
$ mkdir -p bin
$ mkdir -p share/adam-0.23.0-SNAPSHOT-1
$ tar xvfz adam-distribution-spark2_2.11-0.23.0-SNAPSHOT-bin.tar.gz
$ cd adam-distribution-spark2_2.11-0.23.0-SNAPSHOT

$ cp -R * ../share/adam-0.23.0-SNAPSHOT-1/

$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/adam-submit ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/adam-shell ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/find-adam-assembly.sh ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/find-adam-egg.sh ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/find-script-dir.sh ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/find-spark.sh ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/pyadam ../bin/
$ ln -s ../share/adam-0.23.0-SNAPSHOT-1/bin/adamR ../bin/

$ cd ..
$ ./bin/adam-submit --version
Using ADAM_MAIN=org.bdgenomics.adam.cli.ADAMMain
Using spark-submit=/usr/local/bin/spark-submit

       e         888~-_          e             e    e
      d8b        888   \        d8b           d8b  d8b
     /Y88b       888    |      /Y88b         d888bdY88b
    /  Y88b      888    |     /  Y88b       / Y88Y Y888b
   /____Y88b     888   /     /____Y88b     /   YY   Y888b
  /      Y88b    888_-~     /      Y88b   /          Y888b

ADAM version: 0.23.0-SNAPSHOT
Commit: 9d4ba78ad5833d34ec1077f11dda4a9bae4d49ba Build: 2017-08-15
Built for: Apache Spark 2.1.0, Scala 2.11.8, and Hadoop 2.7.3

$ ./bin/adam-shell
Using SPARK_SHELL=/usr/local/bin/spark-shell
...

scala> import org.bdgenomics.adam.rdd.ADAMContext._
import org.bdgenomics.adam.rdd.ADAMContext._

scala> val alignments = sc.loadAlignments("small.sam")
alignments: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD =
RDDBo...

scala> alignments.rdd.count()
res0: Long = 20

scala> :quit
```

File: adam-python/Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ The 'prepare' target installs ADAM's build requirements into the current virtual
 
 The 'develop' target creates an editable install of ADAM and its runtime requirements in the 
 current virtualenv. The install is called 'editable' because changes to the source code 
-immediately affect the virtualenv.
+immediately affect the virtualenv. This target also builds a .egg.
```
Comment (heuermh) [inline adam-python/Makefile]: an .egg?

**Comment by pbashyal-nmdp** [comment]
I'm not able to load files from S3 using `s3a` protocol using `adamR`. I am able to load parquet files using SparkR from S3.

```
[hadoop@ip-10-224-0-159 bin]$ ./adamR
Using SPARKR=/usr/bin/sparkR

R version 3.3.3 (2017-03-06) -- "Another Canoe"
Copyright (C) 2017 The R Foundation for Statistical Computing
Platform: x86_64-redhat-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

Launching java with spark-submit command /usr/lib/spark/bin/spark-submit   "--conf" "spark.kryo.registrator=org.bdgenomics.adam.serialization.ADAMKryoRegistrator" "--conf" "spark.serializer=org.apache.spark.serializer.KryoSerializer" "--conf" "spark.driver.extraClassPath=/home/hadoop/adam/adam-assembly/target/adam-assembly-spark2_2.11-0.23.0-SNAPSHOT.jar" "--jars" "/home/hadoop/adam/adam-assembly/target/adam-assembly-spark2_2.11-0.23.0-SNAPSHOT.jar" "sparkr-shell" /tmp/RtmporBaQu/backend_port6f7a72cd49b6
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
2017-08-15 19:14:08 WARN  Client:66 - Neither spark.yarn.jars nor spark.yarn.archive is set, falling back to uploading libraries under SPARK_HOME.

 Welcome to
    ____              __
   / __/__  ___ _____/ /__
  _\ \/ _ \/ _ `/ __/  '_/
 /___/ .__/\_,_/_/ /_/\_\   version  2.2.0
    /_/


 SparkSession available as 'spark'.
> library(bdg.adam)

Attaching package: ‘bdg.adam’

The following object is masked from ‘package:base’:

    save

> ac <- ADAMContext(spark)
> exacAdam <- "s3a://$AWS_BUCKET/data/ExAC.adam"
> variants <- loadVariants(ac, exacAdam)
2017-08-15 19:17:03 ERROR RBackendHandler:91 - loadVariants on 5 failed
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.api.r.RBackendHandler.handleMethodCall(RBackendHandler.scala:167)
	at org.apache.spark.api.r.RBackendHandler.channelRead0(RBackendHandler.scala:108)
	at org.apache.spark.api.r.RBackendHandler.channelRead0(RBackendHandler.scala:40)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:343)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:336)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:287)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:343)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:336)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:343)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:336)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:343)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:336)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:343)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:643)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:566)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:480)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:442)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2195)
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2702)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2715)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:93)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2751)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2733)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:377)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at org.bdgenomics.adam.rdd.ADAMContext.getFsAndFilesWithFilter(ADAMContext.scala:1340)
	at org.bdgenomics.adam.rdd.ADAMContext.loadAvroSequenceDictionary(ADAMContext.scala:1164)
	at org.bdgenomics.adam.rdd.ADAMContext.loadParquetVariants(ADAMContext.scala:2124)
	at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadVariants$1.apply(ADAMContext.scala:2779)
	at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadVariants$1.apply(ADAMContext.scala:2774)
	at scala.Option.fold(Option.scala:158)
	at org.apache.spark.rdd.Timer.time(Timer.scala:48)
	at org.bdgenomics.adam.rdd.ADAMContext.loadVariants(ADAMContext.scala:2772)
	at org.bdgenomics.adam.api.java.JavaADAMContext.loadVariants(JavaADAMContext.scala:206)
	... 36 more
Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2101)
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2193)
	... 52 more
Error in handleErrors(returnStatus, conn) :
  java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2195)
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2702)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2715)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:93)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2751)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2733)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:377)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at org.bdgenomics.adam.rdd.ADAMContext.getFsAndFilesWithFilter(ADAMContext.scala:1340)
	at org.bdgenomics.adam.rdd.ADAMContext.loadAvroSequenceDictionary(ADAMContext.scala:1164)
	at org.bdgenomics.adam.rdd.ADAMContext.loadParquetVariants(ADAMContext.scala:2124)
	at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loa
```

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by heuermh** [comment]
Tried building from your branch
```
$ git checkout fnothaft/issues/1649-radam-script
$ ./scripts/move_to_spark_2.sh 
$ ./scripts/move_to_scala_2.11.sh 
$ mvn clean install
$ ASSEMBLY_DIR=`pwd`/adam-assembly/target/
$ ASSEMBLY_JAR="$(ls -1 "$ASSEMBLY_DIR" | grep "^adam[0-9A-Za-z\_\.-]*\.jar$" | grep -v javadoc | grep -v sources || true)"
$ export SPARKR_SUBMIT_ARGS="--jars ${ASSEMBLY_DIR}/${ASSEMBLY_JAR} --driver-class-path ${ASSEMBLY_DIR}/${ASSEMBLY_JAR} sparkr-shell"
$ mvn -Pr package
$ ./bin/adamR 

Using SPARKR=/usr/local/bin/sparkR

R version 3.4.2 (2017-09-28) -- "Short Summer"
Copyright (C) 2017 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin16.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

Launching java with spark-submit command /usr/local/Cellar/apache-spark/2.2.0/libexec/bin/spark-submit   "--conf" "spark.kryo.registrator=org.bdgenomics.adam.serialization.ADAMKryoRegistrator" "--conf" "spark.serializer=org.apache.spark.serializer.KryoSerializer" "--conf" "spark.driver.extraClassPath=/Users/heuermh/working/adam/adam-assembly/target/adam-assembly-spark2_2.11-0.23.0-SNAPSHOT.jar" "--jars" "/Users/heuermh/working/adam/adam-assembly/target/adam-assembly-spark2_2.11-0.23.0-SNAPSHOT.jar" "sparkr-shell" /var/folders/m6/4yqn_4q129lbth_dq3qzj_8h0000gn/T//Rtmp3GhhKq/backend_port5e472955adac 
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
2017-11-07 16:43:05 WARN  NativeCodeLoader:62 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2017-11-07 16:43:09 WARN  ObjectStore:568 - Failed to get database global_temp, returning NoSuchObjectException

 Welcome to
    ____              __ 
   / __/__  ___ _____/ /__ 
  _\ \/ _ \/ _ `/ __/  '_/ 
 /___/ .__/\_,_/_/ /_/\_\   version  2.2.0 
    /_/ 


 SparkSession available as 'spark'.
> ac <- ADAMContext(sc)
Error in ADAMContext(sc) : could not find function "ADAMContext"
```
Am I missing something?

File: scripts/release/release.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ git push origin ${branch}
 # do scala 2.10 release
 git checkout -b maint_2.10-${release} ${branch}
 mvn --batch-mode \
-  -P distribution \
+  -P distribution,python \
```
Comment (heuermh) [inline scripts/release/release.sh]: add r to profiles for release

**Comment by fnothaft** [comment]
This should be good to go now. I've squashed down.

**Comment by heuermh** [comment]
Tests fine for me, thank you, @fnothaft!

**Pull request merged**